content/0008-standalone-workflow-execution/rfc.md

```diff
@@ -0,0 +1,294 @@
+# Standalone Workflow Execution (2021-07-07)
+
+## Stakeholders
+
+* **Recommenders:** Kyle Terry
+* **Agreers:** Noah Fontes, Noah Muldavin, Geoff Woodburn
+* **Performers:** Kyle Terry
+* **Inputers:** Rick Lane, Sebastian Prokuski
+* **Deciders:** Kyle Terry, Noah Fontes, Noah Muldavin
+
+## Problem
+
+We don't have a way to run the full stack of services on a development machine
+that allows developers to write code and run workflows against that code. We
+have "feature clusters", but they are largely the same size as production and
+are costly, time-wise, to setup, maintain and keep up to date.
+
+Similarly, we have a core repository that contains the components to run
+workflows, but we don't have a way of easily running those workflows in a
+standalone workflow-engine.
+
+## Summary
+
+This RFC defines an extension to the Relay CLI that wraps existing container and
+Kubernetes management technologies to setup a local Kubernetes cluster that runs
+the necessary Relay services for executing workflows in a standalone
+environment. This RFC also defines a secondary mechanism to automatically run
+workflows after bootstrapping has completed. This will allow the tool to be used
+for setting up and managing a full-stack development environment on a local
+machine.
+
+## Motivation
+
+The ability to run workflows in a standalone system allows developers of
+workflows (Relay users) and Puppet developers who work on the Relay Saas to
+reduce churn when testing new code. For the latter, we want to avoid the wait
+time for building and pushing new changes to a remote testing cluster, which
+requires a cycle through Continuous Integration (CI).
+
+## Product-level explanation
+
+This next section will describe the versions of a tool used to create, manage
+and destroy development clusters.
+
+### V1
+
+We add a new set of commands to the [Relay
+CLI](https://github.com/puppetlabs/relay) used to manage a local Kubernetes
+cluster running the core components that execute workflows.
+
+We also generalize workflow translation and validation code currently in
+relay-api into [relay-sdk](https://github.com/puppetlabs/relay-sdk) so it can be
```

Comment:
I think the right place for this might be `relay-core`, which is where we moved most of the non-Ni-specific SDK code to after we deprecated the nebula-sdk repository. We would consume it from the Relay CLI and the API (which already uses relay-core now). What do you think?